File: String/3146. Permutation Difference between Two Strings/solution.swift

class Solution {
    func findPermutationDifference(_ s: String, _ t: String) -> Int {
        
        var sCount : [Character : Int] = [:]
        var tCount : [Character : Int] = [:]
        
        for (i, c) in s.enumerated(){
            
            sCount[c] = i
        
        }
        
        for (i, c) in t.enumerated(){
            tCount[c] = i
        }
        
        var total = 0
 
        for c in String(s.sorted()) {
            
            if let sIndex = sCount[c], let tIndex = tCount[c] {
                
                let diff = abs(sIndex - tIndex)
                total += diff
                
            }
            
        }
        
        return total
    }
}


// Example 1:
let s : String = "abcde"
let t : String = "edbac"

print(Solution().findPermutationDifference(s, t))

// Example 2:
let s : String = "abc"
let t : String = "bac"

print(Solution().findPermutationDifference(s,t))

File: String/3146. Permutation Difference between Two Strings/solution.py
class Solution:
    def findPermutationDifference(self, s: str, t: str) -> int:
        
        s_count = {}
        t_count = {}

        for i in range(len(s)):
            s_count[s[i]] = i 
            t_count[t[i]] = i 
        

        total = 0

        for c in sorted(s):
            
            diff = abs(s_count[c] - t_count[c])

            total += diff
        
        return total
    

if __name__ == '__main__':

    """ Example 1: """
    s = "abcde"
    t = "edbac"
    print(Solution().findPermutationDifference(s, t))

    """ Example 2: """
    s = "abcde"
    t = "edbac"
    print(Solution().findPermutationDifference(s,t))